docs/dev/developer-guides/transactions/blocks.md

```diff
@@ -0,0 +1,68 @@
+The user signs and sends an operation labeled `ProcessTransactions` to the operator to have their pending transactions processed.
+The operator then includes this operation into the subsequent rollup block, which completes all pending transactions for the sender and changes the on-chain available state to `lastSeenBlockNum(sender) = blockNum`, where `blockNum` is the number of the most recent block. The operator provides witnesses for the updated balances of all users impacted by a rollup block once it has been posted.
+
+
+### Block Numbers: Ethereum vs. zkSync
+
+The structure of L2 blocks is similar to that of L1 blocks. 
+In zkSync, we send to the L1 contract some information about L2 blocks (a commitment), then we produce a proof for the blocks and transmit it back to the L1 contract, which confirms that the proof is accurate.
+
+
+### Ethereum Block Numbers Within zkSync
+
+Accessing block numbers within a zkSync smart contract (i.e., `block. number` in Solidity) is similar to how you would do it on Ethereum. For example, `eth_blockNumber` returns the number of the latest L2 block, and `eth_getBlockByNumber` returns the information about the L2 block by its number.
+
+The **capacity,** which includes the number of transactions and storage writes, of the L1 batch is limited because of two different things:
+
+
+1. For small L1 batches, a prover can produce proofs in an adequate amount of time.
+2. The required gas for the transaction to commit the L1 batch on the L1 contract must not exceed the permitted gas for the L1 block. Because that gas depends on L1 batch size, capacity is constrained.
+
+
+L1 batches are sealed for two reasons:
+
+1. The capacity is reached (well, because we can't send an L1 batch that exceeds capacity).
+2. Timeout has passed. The timeout is between 1 - 24 hours.
+Users require a timeout season to ensure that their transactions become final on L1 at a good time.
+The user would send a few transactions if there weren't a timeout, but the load would be pretty low. As a result, it would take a few days to fill an L1 batch capacity, and the user would have to wait for finality throughout these days, which is not great for the user experience.
+
+### Hashes
+
+You may determine the L2 block's hash by knowing the L2 block number because it is deterministic and equivalent to "keccak256(l2 block number)".
+Every block needs a unique hash, which we mandated for Web3 API compatibility. Additionally, to avoid confusing users, we employ the `keccak hash` function to make hashes appear to be random numbers.
+
+Storage root hash, L1 batch number, and other data are used to determine the hash of the L1 batch. The proofs we supplied to L1 demonstrate that we conduct transactions that cause a state change from the old storage root hash to the committed storage root hash.
+
+### Block Properties
```

Comment:
This paragraph is incorrect, there are many things that we don't have yet on ZkSync, such as block nonce, Beneficiary (because there is only one block producer), mix hash, Transaction root, Receipt root.

I suggest removing it and adding a paragraph `Code block properties`. There you should describe what properties are supported and say that they return properties of L1 batches and not L2 blocks, that's really important. Please ask @StanislavBreadless for the list of the supported properties.

Reply:
@StanislavBreadless could you help with the list of supported properties?

Reply:
@krofax you should take a look at the page about the supported/unsupported opcodes. All the block properties on Ethereum that are not listed as unsupported are supported